src/go/configmanager/config_manager.go

```diff
@@ -132,11 +132,14 @@ func NewConfigManager(mf *metadata.MetadataFetcher, opts options.ConfigGenerator
 	// accessToken is unavailable from imds and --service_account_key must be
 	// set to generate accessToken.
 	// The inverse is not true. We can still use IMDS on GCP when service account key is specified.
-	if mf == nil && opts.ServiceAccountKey == "" {
-		return nil, fmt.Errorf("if flag --non_gcp is specified, flag --service_account_key must be specified")
+	if mf == nil && opts.ServiceAccountKey == "" && opts.EnableApplicationDefaultCredentials == false {
```

Comment:
nit: `&& !opts.EnableApplicationDefaultCredentials`

Reply:
Given that the flag is set to False by default, I am not sure that this is correct? Otherwise - if the flag was not supplied and service account was also not supplied, we would not enter this if statement.

i.e opts.ServiceAccountKey = "" and opts.EnableApplicationDefaultCredentials = False

I did this to try and make it as backwards compatible as possible, so you need to explicitly use this flag for the new behaviour.

Reply:
NVM, I misunderstood :)  - Fixed